test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/YearTest.scala

```diff
@@ -0,0 +1,329 @@
+    assertEquals(Year.of(1), lastBCYear.`with`(ERA, 1))
+
+    assertEquals(firstACYear, firstACYear.`with`(YEAR, 1))
+    assertEquals(Year.of(1), firstACYear.`with`(YEAR_OF_ERA, 1))
+    assertEquals(Year.of(-50), firstACYear.`with`(YEAR, -50))
+    assertEquals(Year.of(50), firstACYear.`with`(YEAR, 50))
+    assertEquals(Year.of(0), firstACYear.`with`(ERA, 0))
+
+    assertEquals(negativeYear, negativeYear.`with`(YEAR, -100))
+    assertEquals(Year.of(0), negativeYear.`with`(YEAR_OF_ERA, 1))
+    assertEquals(Year.of(-50), negativeYear.`with`(YEAR, -50))
+    assertEquals(Year.of(50), negativeYear.`with`(YEAR, 50))
+    assertEquals(Year.of(101), negativeYear.`with`(ERA, 1))
+
+    for (t <- samples) {
+      for (v <- Seq(Long.MinValue, -1000000000L, 1000000000L, Long.MaxValue)) {
+        expectThrows(classOf[DateTimeException], t.`with`(YEAR, v))
+      }
+      for (v <- Seq(Long.MinValue, -1000000001L, 1000000001L, Long.MaxValue)) {
+        expectThrows(classOf[DateTimeException], t.`with`(YEAR_OF_ERA, v))
+      }
+      for (v <- Seq(Long.MinValue, -1L, 2L, Long.MaxValue)) {
+        expectThrows(classOf[DateTimeException], t.`with`(ERA, v))
+      }
+    }
+  }
+
+  @Test def plus(): Unit = {
+    for {
+      t <- samples
+      u <- ChronoUnit.values() if isSupported(u)
+    } {
+      assertSame(t, t.plus(0, u))
+    }
+
+    assertEquals(Year.of(10), lastBCYear.plus(1, DECADES))
+    assertEquals(Year.of(100), lastBCYear.plus(1, CENTURIES))
+    assertEquals(Year.of(1000), lastBCYear.plus(1, MILLENNIA))
+    assertEquals(Year.of(1), lastBCYear.plus(1, ERAS))
+
+    assertEquals(Year.of(11), firstACYear.plus(1, DECADES))
+    assertEquals(Year.of(101), firstACYear.plus(1, CENTURIES))
+    assertEquals(Year.of(1001), firstACYear.plus(1, MILLENNIA))
+
+    assertEquals(lastBCYear, min.plus(999999999L, YEARS))
+    assertEquals(lastBCYear, max.plus(-999999999L, YEARS))
+
+    val values = Seq(Long.MinValue, Int.MinValue.toLong, -1000L, -366L, -365L,
+      -100L, -12L, -10L, -7L, -1L, 0L, 1L, 7L, 10L, 12L, 100L,
+      365L, 366L, 1000L, Int.MaxValue.toLong, Long.MaxValue)
+
+    for {
+      t <- samples
+      v <- values
+    } {
+      testDateTime(t.plus(v, YEARS))(t.plusYears(v))
+      testDateTime(t.plus(v, DECADES))(t.plusYears(Math.multiplyExact(v, 10)))
+      testDateTime(t.plus(v, CENTURIES))(t.plusYears(Math.multiplyExact(v, 100)))
+      testDateTime(t.plus(v, MILLENNIA))(t.plusYears(Math.multiplyExact(v, 1000)))
+    }
+
+    expectThrows(classOf[DateTimeException], firstACYear.plus(1, ERAS))
+    expectThrows(classOf[DateTimeException], lastBCYear.plus(-1, ERAS))
+  }
+
+  @Test def plusYears(): Unit = {
+    for (t <- samples) {
+      assertSame(t, t.plusYears(0))
+    }
+
+    assertEquals(firstACYear, lastBCYear.plusYears(1))
+    expectThrows(classOf[DateTimeException], min.plusYears(-1))
+    expectThrows(classOf[DateTimeException], max.plusYears(1))
+  }
+
```

Comment:
Minus method is tested inherited from `TemporalTest`, it tests the method with a set of sample longs and verifies that it's outcome it's the same as using the `plus` method with negative input:

```scala
@Test def minus(): Unit = {
    for {
      temporal <- samples
      unit <- ChronoUnit.values if temporal.isSupported(unit)
      n <- sampleLongs
    } {
      testDateTime(temporal.minus(n, unit)) {
        if (n != Long.MinValue) temporal.plus(-n, unit)
        else temporal.plus(Long.MaxValue, unit).plus(1, unit)
      }
    }
  }
```

Didn't override it as none of the other `TemporalTest` implementors do so, they just test the concrete `minusXXX` methods (e.g.: `minusYears`)